llama-index-core/llama_index/core/indices/vector_store/base.py

```diff
@@ -345,10 +345,15 @@ def delete_nodes(
             node_ids (List[str]): A list of node_ids from the nodes to delete
 
         """
-        raise NotImplementedError(
-            "Vector indices currently only support delete_ref_doc, which "
-            "deletes nodes using the ref_doc_id of ingested documents."
-        )
+        # delete nodes from vector store
+        self._vector_store.delete_nodes(node_ids, **delete_kwargs)
+
+        # delete from docstore only if needed
+        if (
+            not self._vector_store.stores_text or self._store_nodes_override
+        ) and delete_from_docstore:
+            for node_id in node_ids:
+                self._docstore.delete_document(node_id, raise_error=False)
```

Comment:
@logan-markewich - does this look more or less correct to you? Also, we have a `_delete_node()` method that currently is not implemented. We could outsource this logic to that method and just call on it here in a loop.

Reply:
I think this looks fine to me, no need to loop if we have the batch operation in place 👍🏻

Reply:
thanks, stgm